src/Features/Core/Portable/ReplaceMethodWithProperty/ReplaceMethodWithPropertyCodeRefactoringProvider.cs

```diff
@@ -21,468 +21,466 @@
-        {
-            var allReferenceDocuments = getReferencesByDocument.Concat(setReferencesByDocument).Select(g => g.Key).Distinct();
-            foreach (var referenceDocument in allReferenceDocuments)
-            {
-                cancellationToken.ThrowIfCancellationRequested();
+        return updatedSolution;
+    }
 
-                updatedSolution = await UpdateReferencesInDocumentAsync(
-                    propertyName, nameChanged, updatedSolution, referenceDocument,
-                    getReferencesByDocument[referenceDocument],
-                    setReferencesByDocument[referenceDocument],
-                    cancellationToken).ConfigureAwait(false);
-            }
+    private static async Task<Solution> UpdateReferencesAsync(Solution updatedSolution, string propertyName, bool nameChanged, ILookup<Document, ReferenceLocation> getReferencesByDocument, ILookup<Document, ReferenceLocation> setReferencesByDocument, CancellationToken cancellationToken)
+    {
+        var allReferenceDocuments = getReferencesByDocument.Concat(setReferencesByDocument).Select(g => g.Key).Distinct();
+        foreach (var referenceDocument in allReferenceDocuments)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
 
-            return updatedSolution;
+            updatedSolution = await UpdateReferencesInDocumentAsync(
+                propertyName, nameChanged, updatedSolution, referenceDocument,
+                getReferencesByDocument[referenceDocument],
+                setReferencesByDocument[referenceDocument],
+                cancellationToken).ConfigureAwait(false);
         }
 
-        private static async Task<Solution> UpdateReferencesInDocumentAsync(
-            string propertyName,
-            bool nameChanged,
-            Solution updatedSolution,
-            Document originalDocument,
-            IEnumerable<ReferenceLocation> getReferences,
-            IEnumerable<ReferenceLocation> setReferences,
-            CancellationToken cancellationToken)
-        {
-            var root = await originalDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        return updatedSolution;
+    }
 
-            var editor = new SyntaxEditor(root, originalDocument.Project.Solution.Services);
-            var service = originalDocument.GetRequiredLanguageService<IReplaceMethodWithPropertyService>();
+    private static async Task<Solution> UpdateReferencesInDocumentAsync(
+        string propertyName,
+        bool nameChanged,
+        Solution updatedSolution,
+        Document originalDocument,
+        IEnumerable<ReferenceLocation> getReferences,
+        IEnumerable<ReferenceLocation> setReferences,
+        CancellationToken cancellationToken)
+    {
+        // No point updating references in generated code, it will just get regenerated based on the changes to actual code.
+        if (originalDocument.Id.IsSourceGenerated)
```

Comment:
```suggestion
        if (originalDocument is SourceGeneratedDocument)
```

Since I'm not sure if we want to treat that bit on the ID as more of an implementation detail of the ID.